eth/blockwatch/block_watcher.go

```diff
@@ -281,28 +284,45 @@ func (w *Watcher) addLogs(header *MiniHeader) (*MiniHeader, error) {
 func (w *Watcher) getMissedEventsToBackfill(ctx context.Context) ([]*Event, error) {
 	events := []*Event{}
 
+	var (
+		startBlockNum          int
+		latestRetainedBlockNum int
+		blocksElapsed          int
+	)
+
 	latestRetainedBlock, err := w.stack.Peek()
 	if err != nil {
 		return events, err
 	}
-	// No blocks stored, nowhere to backfill to
-	if latestRetainedBlock == nil {
-		return events, nil
-	}
+
 	latestBlock, err := w.client.HeaderByNumber(nil)
 	if err != nil {
 		return events, err
 	}
-	blocksElapsed := big.NewInt(0).Sub(latestBlock.Number, latestRetainedBlock.Number)
-	if blocksElapsed.Int64() == 0 {
+	latestBlockNum := int(latestBlock.Number.Int64())
+
+	if latestRetainedBlock != nil {
+		latestRetainedBlockNum = int(latestRetainedBlock.Number.Int64())
+		// Events for latestRetainedBlock already processed, start at latestRetainedBlock + 1
+		startBlockNum = latestRetainedBlockNum + 1
+		blocksElapsed = latestBlockNum - startBlockNum
```

Comment:
This line is repeated in this clause and the next clause - can we move it after this if/else if/else block?

Reply:
fixed in be50d19d